### PR TITLE
feat: Declare formIssue schema

### PR DESCRIPTION
**Pull request by mayying**
Define formIssue schema. 

image

Context: Once public user reports an issue, formSG will store the reported issue in a new collection, named `formIssue`. The queries used for this collection are:
1. `db.getCollection("formIssue").save()`
 To save new reported issue into the collection (Note that email is an optional field from the UI)
2. `db.getCollection("formIssue").countDocuments({ "formId": ObjectId(""), "_id":{$gte:ObjectId(""),$lt:ObjectId("")}}).limit(1)` 
 To search if there're other prior issues reported per form per day (to decide if an email notification is needed)
3.  `db.getCollection("formIssue").find({"formId": ObjectId("")}).sort().skip().limit()`
 To lookup issue and return to admin dashboard

### CODE REVIEW

File: shared/types/form/form_issue.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { FormDto } from './form'
+
+/**
+ * Typing for individual form issue
+ */
+export type FormIssueBase = {
+  formId: FormDto['_id']
+  issue: string
+  email?: string
+  created?: Date
```
Comment (ghost) [inline shared/types/form/form_issue.ts:10]: Curious, why would created be an optional field?

**Comment by sebastianwzq** [comment]
LGTM but I will let more experienced folks review/comment

File: shared/types/form/form_issue.ts (hunk 1 above)
Comment (timotheeg) [reply shared/types/form/form_issue.ts:10]: I think this is a common problem for creation, where the dataset we supply does not contain a `created` date (that gets generated by mongoose), but thereafter all reads will have a `created` field. Ideally we'd have a better way to declare the create/read distinction 🤔

File: src/app/models/__tests__/form_issue.server.model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('form_issue.server.model', () => {
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => await dbHandler.clearDatabase())
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('Schema', () => {
+    const DEFAULT_PARAMS: IFormIssue = {
+      formId: new ObjectId(),
+      issue: 'I am 200 years old but your age drop down is only cap at 100',
```
Comment (timotheeg) [inline src/app/models/__tests__/form_issue.server.model.spec.ts:20]: Haha, great "problem" 😄

File: src/types/form_issue.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { FormIssueBase } from 'shared/types'
+import type { Merge } from 'type-fest'
+
+import { IFormSchema } from './form'
+
+export type IFormIssue = Merge<FormIssueBase, { formId: IFormSchema['id'] }>
+
+export interface IFormIssueSchema extends IFormIssue, Document {
+  created?: Date
+  lastModified?: Date
```
Comment (timotheeg) [inline src/types/form_issue.ts]: Is it necessary to redeclare these 2 properties? 🤔 

`IFormIssueSchema` extends `IFormIssue`, itself created from `FormIssueBase`, which already contains the 2 fields.

**Review by timotheeg — APPROVED** [review]
LGTM 👍 Just one nit question

File: src/types/form_issue.ts (hunk 3 above)
Comment (mayying) [reply src/types/form_issue.ts]: Honestly, all other schemas are defined the same way, where the Base class also has a created and lastModified. But you are right, it seems to be just duplicated. I have tested it and the flow still works

**Pull request merged**